File: src/main/java/io/github/glassmc/loader/launch/Wrapper.java
package io.github.glassmc.loader.launch;

import java.lang.reflect.InvocationTargetException;
import java.lang.reflect.Method;

public class Wrapper {

    public static void launch(String target, String[] args) {
        try {
            Class<?> targetClass = Class.forName(target);
            Method targetMethod = targetClass.getMethod("main", String[].class);
            targetMethod.invoke(null, (Object) args);
        } catch (ClassNotFoundException | NoSuchMethodException | IllegalAccessException | InvocationTargetException e) {
            e.printStackTrace();
        }
    }

}

File: core/src/main/kotlin/ml/glassmc/loader/launch/Wrapper.kt
package ml.glassmc.loader.launch

class Wrapper {

    fun launch(target: String, args: Array<String>) {
        val targetClass = Class.forName(target)
        val targetMethod = targetClass.getMethod("main", Array<String>::class.java)
        targetMethod.invoke(null, args)
    }

}